plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -438,15 +440,17 @@ public synchronized void createTable(
             Table table,
             PrincipalPrivileges principalPrivileges,
             Optional<Path> currentPath,
+            Optional<List<String>> files,
```

Comment:
It can I guess. I made it `Optional` to be consistent with `currentPath` being optional. So this is obvious that for given code path `files` do not make sense to be set. 
I can change it to be List and pass empty list, but Optional feels more natural to me.

Reply:
It just seems a little confusing in what cases the list of files is set. What are the downsides of always passing the list of files written?

Reply:
I do when I have them. But there are cases when I do not have them at hand (register partition for existing directory) or they just do not make sense (create table call for partitioned table).